bindgen/ir/location/Location.h

```diff
@@ -0,0 +1,9 @@
+#ifndef SCALA_NATIVE_BINDGEN_LOCATION_H
+#define SCALA_NATIVE_BINDGEN_LOCATION_H
+
+class Location {
+  public:
+    virtual ~Location() = default;
+};
```

Comment:
I think it would be better to have a single `Location` class with the include file path, `isInMainFile` and position in the file for error reporting etc. It sounds like you are considering that. If we put the position there it would also render much (or all) of the `LocationManager` obsolete because the `TreeVisitor` will just generate a new `Location` every time. The mapping from file path to Scala module path can then be done in the generator.